pixl_dcmd/src/pixl_dcmd/_database.py

```diff
@@ -62,6 +62,30 @@ def get_uniq_pseudo_study_uid_and_update_db(
         return UID(pseudo_study_uid, validation_mode=pydicom.config.RAISE)
 
 
+def get_pseudo_patient_id_and_update_db(
+    project_slug: str, study_info: StudyInfo
+) -> UID:
+    """
+    Checks if record (by slug and study info) exists in the database,
+    gets the pseudo_paitent_id if it is not None or records a new, unique one.
+    Returns the pseudo_paitent_uid.
+    """
+    PixlSession = sessionmaker(engine)
+    with PixlSession() as pixl_session, pixl_session.begin():
+        existing_image = get_unexported_image(
+            project_slug,
+            study_info,
+            pixl_session,
+        )
+
+        if existing_image.pseudo_patient_id is None:
+            pseudo_patient_id = generate_uid()
+        else:
+            pseudo_patient_id = existing_image.pseudo_patient_id
+
+        return UID(pseudo_patient_id, validation_mode=pydicom.config.RAISE)
```

Comment:
This shouldn't return a UID, but we should set the DICOM data to be the `pseudo_patient_id` value at the end of this. If its useful for testing we can have it return the `pseudo_patient_id`, but doesn't need to

Reply:
get_pseudo_patient_id_and_update_db just returns existing_image.pseudo_patient_id: [0244bcd](https://github.com/UCLH-Foundry/PIXL/pull/484/commits/0244bcdd925b41b4d734e3f7f6210e0747530bb4)

Reply:
ah I don't think this is resolved because we're not updating the DICOM `dataset` patient identifier tag with the output of this function